packages/api/portfolioDrafts/application/createApplicationStep.ts

```diff
@@ -0,0 +1,104 @@
+import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { ApplicationStep } from "../../models/ApplicationStep";
+import { APPLICATION_STEP } from "../../models/PortfolioDraft";
+import { dynamodbDocumentClient as client } from "../../utils/dynamodb";
+import {
+  DATABASE_ERROR,
+  NO_SUCH_PORTFOLIO_DRAFT,
+  PATH_PARAMETER_REQUIRED_BUT_MISSING,
+  REQUEST_BODY_EMPTY,
+  REQUEST_BODY_INVALID,
+} from "../../utils/errors";
+import { ApiSuccessResponse, ErrorResponse, ErrorStatusCode, SuccessStatusCode } from "../../utils/response";
+import { isApplicationStep, isValidJson } from "../../utils/validation";
+import { ErrorCodes, ValidationError } from "../../models/Error";
+
+/**
+ * Submits the Application Step of the Portfolio Draft Wizard
+ *
+ * @param event - The POST request from API Gateway
+ */
+export async function handler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
+  if (!event.body) {
+    return REQUEST_BODY_EMPTY;
+  }
+
+  const portfolioDraftId = event.pathParameters?.portfolioDraftId;
+
+  if (!portfolioDraftId) {
+    return PATH_PARAMETER_REQUIRED_BUT_MISSING;
+  }
+
+  if (!isValidJson(event.body)) {
+    return REQUEST_BODY_INVALID;
+  }
+  const requestBody = JSON.parse(event.body);
+
+  // remove this, and function?
+  if (!isApplicationStep(requestBody)) {
+    return REQUEST_BODY_INVALID;
+  }
+  const now = new Date().toISOString();
+  const applicationStep: ApplicationStep = requestBody;
+
+  // NOTE This is absolutely not want we want to do long term but I am just trying to meet the ACs as simply as possible
+  const error = validate(applicationStep);
+  if (error) {
+    return new ErrorResponse({ code: error.code, message: error.message }, ErrorStatusCode.BAD_REQUEST);
+  }
+
+  const command = new UpdateCommand({
+    TableName: process.env.ATAT_TABLE_NAME ?? "",
+    Key: {
+      id: portfolioDraftId,
+    },
+    UpdateExpression: "set #portfolioVariable = :application, updated_at = :now",
+    ExpressionAttributeNames: {
+      "#portfolioVariable": APPLICATION_STEP,
+    },
+    ExpressionAttributeValues: {
+      ":application": applicationStep,
+      ":now": now,
+    },
+    ConditionExpression: "attribute_exists(created_at)",
+  });
```

Comment:
Do we need the `ReturnValues: "ALL_NEW"`? The default value is `NONE` according to the [DynamoDB docs](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_UpdateItem.html). 

I have tried changing this value for the portfolioStep on my cdk deploys, but still got the same response. Not sure if I was doing something wrong though.

Reply:
Good question.  In this case, I believe this is satisfying [this part of the 201 response](https://github.com/dod-ccpo/atat-web-api/blob/c95b728959676097b7aafe76f55a7f210a864c38/atat_provisioning_wizard_api.yaml#L338) in the API spec that says the content (body) should be a JSON representation of an `ApplicationsStep`.

>           content:
>            application/json:
>              schema:
>                $ref: '#/components/schemas/ApplicationStep'
>              examples:
>                  ApplicationStepEx:
>                    $ref: '#/components/examples/ApplicationStepEx'

We seem to be doing that consistently across the board for the POST functions.  Apart from being in the spec, at present I think it's also immensely helpful to the testers.
![image](https://user-images.githubusercontent.com/77642966/133360953-c44733e8-fb86-4b88-9788-a74130fe9ae8.png)

Reply:
ALL_NEW - Returns all of the attributes of the item, as they appear after the UpdateItem operation. We are mainly using this to validate that the application_step was actually added to the DB, so it should be fine to leave it in.